contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -190,15 +220,27 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
     }
 
     /**
-     * @dev See {acceptDefaultAdminTransfer}.
+     * @dev See {scheduleDefaultAdminDelayChange}.
      *
      * Internal function without access restriction.
      */
-    function _acceptDefaultAdminTransfer() internal virtual {
-        require(_hasDefaultAdminTransferDelayPassed(), "AccessControl: transfer delay not passed");
-        _revokeRole(DEFAULT_ADMIN_ROLE, defaultAdmin());
-        _grantRole(DEFAULT_ADMIN_ROLE, pendingDefaultAdmin());
-        _resetDefaultAdminTransfer();
+    function _scheduleDefaultAdminDelayChange(uint48 newDefaultAdminDelay) internal virtual {
+        require(defaultAdminTransferSchedule() != 0, "AccessControl: default admin transfer pending");
+
+        uint48 currentDefaultAdminDelay = defaultAdminDelay();
+
+        // Schedules the current delay if the delay is increased.
+        // Otherwise, wait the difference between current and new delay to guarantee the delay change schedule + a default admin change
+        // is effectively the current delay. For example, if delay is reduced from 10 days to 3 days, it's needed to wait 7 days
+        // before starting the new 3 days delayed transfer suming up to 10 days, which is the current delay.
+        uint48 changeDelay = newDefaultAdminDelay >= currentDefaultAdminDelay
+            ? newDefaultAdminDelay
```

Comment:
This seems like a mistake. It should be the current delay, no?
```suggestion
            ? currentDefaultAdminDelay
```

Reply:
The mistake was in the comments. I just updated. Let me explain the rationale:

If a user sets an accidental high delay and wait for only the previous delay, there's a risk if they begin an admin transfer and got locked. This is a way to ensure they didn't set a high value, because they will always need to wait for it, and it's also cancellable

Reply:
Fixed in [this commit](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4079/commits/ff7a4d63c9f77ad45ec8c29e9f9438370045ae83)